<a href="https://colab.research.google.com/github/Sowmya-2348562/ML-lab/blob/main/2348562_Prg3b.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [ ]:
import pandas as pd
from sklearn.model_selection import train_test_split
from sklearn.linear_model import LogisticRegression
from sklearn.metrics import accuracy_score, precision_score, recall_score, classification_report, confusion_matrix
from sklearn.preprocessing import StandardScaler


In [ ]:
df=pd.read_csv('/content/Program_3b_dataset.csv')

In [ ]:
# Define predictors and target variable
X = df[['variance', 'skewness', 'curtosis', 'entropy']]
y = df['class']


In [ ]:
# Split the dataset into training and testing sets
X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.2, random_state=42)


In [ ]:
# Standardize the predictors
scaler = StandardScaler()
X_train = scaler.fit_transform(X_train)
X_test = scaler.transform(X_test)


In [ ]:
# Build the logistic regression model
logistic_model = LogisticRegression()
logistic_model.fit(X_train, y_train)

LogisticRegression()

In [ ]:

# Make predictions
y_pred = logistic_model.predict(X_test)


In [ ]:
# Calculate accuracy, precision, and recall
accuracy = accuracy_score(y_test, y_pred)
precision = precision_score(y_test, y_pred)
recall = recall_score(y_test, y_pred)

In [ ]:
# Display Classification Report and Confusion Matrix
print("Classification Report:")
print(classification_report(y_test, y_pred))

print("Confusion Matrix:")
print(confusion_matrix(y_test, y_pred))

Classification Report:
              precision    recall  f1-score   support

           0       0.99      0.97      0.98       148
           1       0.97      0.98      0.98       127

    accuracy                           0.98       275
   macro avg       0.98      0.98      0.98       275
weighted avg       0.98      0.98      0.98       275

Confusion Matrix:
[[144   4]
 [  2 125]]


Inference:
Based on the classification report and confusion matrix, the logistic regression model performs very well on the data. The accuracy is 98%, which means that the model correctly predicts the class label for 98% of the data points. The precision and recall for both classes are also very high, meaning that the model is able to accurately identify both positive and negative cases. The confusion matrix shows that the model only misclassified 4 out of 275 data points, which is a very small error rate